### DISCUSSION TITLE
We’re Listening – Help Us Make GitLens Even Better

### DISCUSSION THREAD

**Discussion by SpaceMerlin33** [gitkraken/vscode-gitlens#3668] [Feedback]
At GitLens, we’re dedicated to continuously improving your experience, and _your feedback_ is key to shaping the future of our free and Pro features. Whether it’s frustrations, complaints, or just something you feel we should change, we want to hear directly from you. 

Communicate directly with our engineering team – we’re listening, share your thoughts with us here.

_*For technical issues or bugs please [open an issue here](https://github.com/gitkraken/vscode-gitlens/issues)_

**Comment by dEhiN** [comment]
It would be nice to have the option to start a GitLens Pro Trial when creating an account and authorizing access to GitHub. I recently started with GitLens and, while exploring its features - including using the YouTube tutorial [here](https://www.youtube.com/watch?v=oJdlGtsbc3U&ab_channel=GitKraken) - I decided to connect GitLens to my GH account. I was a little disappointed to see that, as part of the authorization, a GitLens Pro Trial was started for me, thereby giving me 7 days to try all the Pro features.

In my case, I currently only have public-facing repos, so I'm not worried about losing access to things such as the Commit Graph. However, it's precisely for this reason that I don't want to start a Pro Trial yet. I think the choice should be presented to the user when connecting GitLens to GH - would you like to start a Pro Trial now? If the user chooses no, they then still have the ability to initiate a trial at a later date, perhaps once they have a private repo they want/need to work on.

**Reply by justinrobots** [reply]
Hi @dEhiN - thanks for sharing your feedback! We recently extended the trial period for GitLens Pro from 7 to 14 days to give users more time to explore Pro features. We also periodically allow trials to be reset so you can check out Pro features again when there are updates or when the timing feels right.

We hope these changes give you plenty of flexibility to try Pro features whenever works best for you.

Appreciate your input—it's super helpful for shaping GitLens!

**Reply by dEhiN** [reply]
@justinrobots Thanks for the reply! I'm glad to hear the time has been extended. I'm working on a solo project myself but I still am finding GitLens a lot of help.

In fact, I have an older laptop that I sometimes work on, and I have deactivated most of my extensions to not tax the system. This included GitLens, but I'm used to it now from my desktop that I miss it! Especially the inline blame.

**Comment by adaPOSai** [comment]
❤️❤️❤️

**Comment by emike108** [comment]
Reposting here to see if I can get a response, thank you!
https://github.com/gitkraken/vscode-gitlens/discussions/3721#discussioncomment-12543060

**Comment by SaeedDev94** [comment]
Please add rollback option in explorer (context menu) for file and directory  
Also in confirm dialog we should be able to select or deselect directories and files before rollback.  
Currently we have to use vscode "Source Control" for rollback, IMO rollback directly in (file) explorer (without changing tab) is so much easier!  
Thanks :)

**Comment by CharlieDixon** [comment]
The main issues I'm having are performance-related. I've noticed that with the default settings, if I have any medium to large repositories open (e.g. hundreds of branches with tens of thousands of commits) both GitKraken and GitLens quickly become unusable and I have to force quit them.

Despite the many great-sounding features I've read about for the launchpad, I avoid using it altogether as it either shows a spinning wheel or crashes the application if I navigate to that tab in either app.

With repositories that have >100k commits & >10k branches, both tools become unusable in my experience, or at least not on my hardware. I am using the latest versions for both applications on a Macbook M2 Pro w/16GB of RAM & have macOS Sequoia 15.4.1 installed. The main symptom I notice quite quickly after booting my laptop and switching between a few different repos, is that it quickly begins to overheat. In `Activity Monitor` I've seen the `kernel_task` process go as high as 850% in an effort to cool the machine/protect it from damage.

As a result, I've recently decided to remove GitLens altogether and selectively use GitKraken for repos within a size or complexity range it manages with on my system or smaller projects on my personal laptop etc and this has helped a lot with the overall reported CPU usage & heat.

If there are any tips you could point me to which might help with either that'd be great, as the feature set is superb when it works/is performant. I am using GitKraken (Business) and was using GitLens (Community) edition

**Reply by justinrobots** [reply]
Hey @CharlieDixon , sorry to hear that you're encountering performance issues with large repositories in GitLens. We just released [GitLens 17.1](https://help.gitkraken.com/gitlens/gitlens-release-notes-current/) today with many Git performance improvements and I'm curious to hear if you notice any difference after updating.

There is now a setting (`gitlens.advanced.commits.delayLoadingFileDetails`) to delay loading full commit file details until required to improve performance for large repositories. If upgrading to 17.1 on it's own doesn't resolve your issues, try that setting and let us know if you see a difference.

If you're still encountering issues, we'd love to set up some time to chat about your specific case. You can join our [Community Slack](https://slack.gitkraken.com/) and message me (`justin_roberts`) to set that up if you're willing.

**Reply by eamodio** [reply]
@CharlieDixon Can you try out the latest release (or pre-release) to see if you see a difference. In GL 17.1 we spent a lot of time optimizing performance and especially with large repositories. If you are still seeing slowness, there is also a new setting `gitlens.advanced.commits.delayLoadingFileDetails` that you can enable that should help things even more on large repos (could have a negative on smaller repos).

If the new version doesn't resolve the issues, please open a new issue and provide logs, as we'd love to get to the bottom of it.

You can also join our Slack community: https://www.gitkraken.com/join-slack-community as we'd love to hear more about your experience with GitLens and how we can improve.

**Reply by justinrobots** [reply]
Apologies for the nearly duplicate responses. @eamodio  and I must be on the same wavelength today 😄

**Comment by babakfp** [comment]
Me want this: https://github.com/gitkraken/vscode-gitlens/discussions/3578

**Comment by thomaslevesque** [comment]
GitLens is aggressively pushing "Pro" features that I don't need.
I have already disabled pro features (using the "Hide Pro Features" command), but I get this dialog every time I do an interactive rebase:
<img width="550" height="166" alt="image" src="https://github.com/user-attachments/assets/3dde5fd7-9940-4576-906f-a1063516a4da" />
The interactive rebase tool isn't a pro feature, so why am I seeing this? I don't need the pro features, and I don't want to keep being nagged about them.

Note that this is a recent evolution. I had never been bothered about Pro features until maybe... 2 months ago?

**Reply by eamodio** [reply]
Woah -- you are getting that when just opening the rebase editor? If so that is a nasty bug and NOT intentional.

**Reply by thomaslevesque** [reply]
Yup. I just type `git rebase -i main` in the CLI, which opens VSCode since it's my git editor.
Glad to know it's not intentional!

**Reply by eamodio** [reply]
Definitely not -- What version of GitLens are you using? 17..8.1 shouldn't exhibit this behavior (though we've found some other cases where this could happen when not wanted -- which we are fixing ASAP).

**Reply by thomaslevesque** [reply]
I have 17.8.1, but I just realized the extension had a pending restart, so I might have been using an older version. After restarting the extension, I no longer see this issue. Thanks!

**Comment by josephbradfield** [comment]
I have been a paid user of gitlens for some time now. i have never been able to get my PR workflow setup in launchpad.
all i ever see is "All Done take a vacation" but i have open PRs on the repo.  The Commit graph shows tons of things do i have to use github "issues" to work on them our org uses jira to manage tickets. Really want to love GitLens more! Help me help you

**Comment by CZNetworks** [comment]
> At GitLens, we’re dedicated to continuously improving your experience, and _your feedback_ is key to shaping the future of our free and Pro features. Whether it’s frustrations, complaints, or just something you feel we should change, we want to hear directly from you.
> 
> Communicate directly with our engineering team – we’re listening, share your thoughts with us here.
> 
> _*For technical issues or bugs please [open an issue here](https://github.com/gitkraken/vscode-gitlens/issues)_

**Comment by ZeeshanAhmadAIO** [comment]
GitLens VS Code extension is eating up all the memory in macOS if I have a large monorepo. This is a deal-breaker for me.

https://github.com/gitkraken/vscode-gitlens/issues/5504
